frontend/src/index.test.js

```diff
@@ -1,37 +1,2524 @@
-import { render, screen } from "@testing-library/react";
-import {setupLandingPage} from "./setupTests";
-
-describe('MyComponent tests', () => {
-  let getByTestId;
-
-  test('empty test', () => {
-    return;
-  })
-  /*
-  beforeAll(() => {
-    // const renderResult = setup();
-    // getByTestId = renderResult.getByTestId;
-    getByTestId = setupLandingPage().getByTestId;
-  });
-
-  test('component has the correct title prop', () => {
-    const element = getByTestId('app-name'); // Adjust this to your actual test ID or query method
-    expect(element).toHaveTextContent('TermHub');
-  });
-
-   */
-  /*
-  test("renders the app starting with BrowserRouter and AppWrapper", () => {
-    // let linkElement = screen.getAllByText(/TermHub/i)[0];
-    // expect(linkElement).toBeInTheDocument();
-    // linkElement = screen.getByText(/CSet Search/i);
-    // expect(linkElement).toBeInTheDocument();
-  });
-   */
-
-  // Additional tests here
+// TODO: Test cases for: makeGraph(), setDescendantStats(), getIndentedTreeNodes()?, cset comparison page line 144?
+// TODO: Inputs: real inputs in addition to wholeGraphEdge
+import {test, expect} from '@jest/globals';
+
+import {GraphContainer, makeGraph} from "./state/GraphState";
+
+
+// UTILS
+/* Used to convert input to be same as graphology serialization (all strings). */
+function convertToArrayOfStrings(matrix) {
+  var stringMatrix = [];
+  for (var i = 0; i < matrix.length; i++) {
+    var stringRow = [];
+    for (var j = 0; j < matrix[i].length; j++) {
+      stringRow.push(matrix[i][j].toString());
+    }
+    stringMatrix.push(stringRow);
+  }
+  return stringMatrix;
+}
+
+// VARS
+// diagramCase: last copied from test_graph.py test_get_missing_in_between_nodes(): 2024/02/11
+let diagramCase = [
```

Comment:
## Static variables VS external JSON
I tried to save these as JSON and load from file system but I got error:
```
(node:55454) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension. (Use `node --trace-warnings ...` to show where the warning was created)
```

**The variables here are**
Cases:
- diagram case
- single small
- many small

Formats:
- edges only
- `concept_graph()` return
- `GraphContainer` input: `graphData`